enhancements/20251009-nic-hotplug.md

```diff
@@ -0,0 +1,133 @@
+
+### User Experience In Detail
+
+The following steps outline the user workflow for managing network interfaces on a running VM:
+
+1.  Navigate to the details page of a running VM.
+2.  On the **Networks** tab, interfaces that support hot-unplug will display a corresponding action button (e.g., "Delete" or "Unplug").
+3.  To hot-plug an interface, the user can click the kebab menu (three vertical dots) on either the VM details page or the VM list page and select the **Add Network Interface** option. A dialog will appear, prompting the user to specify an interface name and select a network from a dropdown list.
+
+### API changes
+
+Two new API actions will be introduced for the `VirtualMachine` resource.
+
+**Add NIC**
+```
+POST /v1/harvester/vm/<vm>?action=addNic
+Content-Type: application/json
+
+{
+  "interfaceName": "dynif1",
+  "networkName": "default/vmnet1",
+  "bindingMethod": "bridge"  // or "sriov"
+}
+```
+
+**Remove NIC**
+```
+POST /v1/harvester/vm/<vm>?action=removeNic
+Content-Type: application/json
+
+{
+  "interfaceName": "dynif1"
+}
+```
+
+## Design
+
+
+### Implementation Overview
+
+#### Satisfy KubeVirt Hot-plug Network Interface Requirements
+
+According to the KubeVirt documentation[^1], two requirements must be met:
+
+1.  Multus CNI must be deployed using a [thick plugin](https://github.com/k8snetworkplumbingwg/multus-cni/blob/master/docs/thick-plugin.md) architecture.
+2.  The Multus Dynamic Networks Controller must be installed.
+
+As per the RKE2 documentation, these requirements can be met by applying the appropriate `HelmChartConfig` via the Harvester installer.
+
+```yaml
+# pkg/config/templates/rancherd-23-multus-config.yaml
+resources:
+- apiVersion: helm.cattle.io/v1
+  kind: HelmChartConfig
+  metadata:
+    name: rke2-multus
+    namespace: kube-system
+  spec:
+    valuesContent: |-
+      thickPlugin:
+        enabled: true
+      dynamicNetworksController:
+        enabled: true
+```
+
+#### Network Interface Hot-plug Actions
+
+The API server will provide two new actions, `addNic` and `removeNic`, for `VirtualMachine` resources.
+
+-   `addNic` will patch the `VirtualMachine` resource by adding new elements to `spec.template.spec.domain.devices.interfaces` and `spec.template.spec.networks`. The interface `model` will be hardcoded to `virtio`, and only `bridge` and `SR-IOV` binding methods will be permitted.
+
+-   `removeNic` will patch the `VirtualMachine` resource by setting the `state` of the specified interface in `spec.template.spec.domain.devices.interfaces` to `absent`. This action will only be permitted for interfaces using the `bridge` binding method.
+
+### Test plan
+
+#### Regression tests
+
+Since this enhancement migrates the Multus CNI from a thin to a thick plugin architecture, all existing network functionalities, including the storage network, must be thoroughly tested to prevent regressions.
+
+#### Functional tests
```

Comment:
As suggested by @ihcsim during meeting, we could try hot-plug and hot-unplug multiple times and observe the behavior.